### PR TITLE
[ADD] ActivateUserView and Email Support

### PR DESCRIPTION
**Pull request by choiboii**
---
## Checklist:

- [x] Added new function
- [x] Review requested
- [ ] Refactoring
- [ ] Debugging
- [x] Convention fixing

---

## Description:
- ActivateUserView to activate users after signing up
- SMTP email server used to send emails (using Sendinblue email server)
- Adjusted LoginView to prevent signing up before activating account
- Added tests + fixed previous tests to match current moadels/views

---

### Tests run in test.py (2 new tests added + loginview + passwordchangeview + doctorlistview tests fixed):
activate_user_tests

---

### Process of signing up and activating account:
**Successful signup:**

signup_email_success

<br>

**Activation Email:**

email_signup

<br>

**Login prior to activating account:**

login_view_pre_activation

<br>

**Sucess in activating account (both Postman and HTML work)**

success_activation

<br>

success_activation_html

<br>

**Successful login after activation**

login_success_after_activation

<br>

**Error that could appear if given an invalid link**

invalid_link(token or uid wrong)

<br>

**Error if the user's account has already been activated**

user_already_activated

### CODE REVIEW

File: templates/acc_active_email.html [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{% autoescape off %}  
+Hi {{ name }},  
+Welcome to VOIDOC!
+Please click on the link to confirm your registration,
+http://{{ domain }}{% url 'activate' uid=uid token=token %}
```
Comment (jasonkang14) [inline templates/acc_active_email.html]: there should be a way to hide this url from the email body. if yo add a button to access the link or a simple hyper text will improve user experience

File: users/models.py [hunk 2]
```diff
@@ -39,6 +39,7 @@ class CustomUser(AbstractBaseUser):
     is_doctor    = models.BooleanField(default=False, null=True)
     date_joined  = models.DateTimeField(auto_now_add=True)
     is_superuser = models.BooleanField(default=False)
+    is_active    = models.BooleanField(default=True)
```
Comment (jasonkang14) [inline users/models.py:42]: why are you adding `is_active`?

File: users/views.py [hunk 3]
```diff
@@ -4,9 +4,15 @@
...
 from django.forms           import ValidationError
+from django.core            import mail
 from django.core.validators import validate_email
 from django.core.exceptions import ObjectDoesNotExist
 from django.contrib.auth    import login, authenticate
+from django.contrib.auth.tokens import default_token_generator
+from django.contrib.sites.shortcuts import get_current_site  
+from django.template.loader import render_to_string  
+from django.utils.encoding import force_bytes, force_str
+from django.utils.http import urlsafe_base64_encode, urlsafe_base64_decode
```
Comment (jasonkang14) [inline users/views.py:15]: did you end up finding out what these functions do?

File: templates/acc_active_email.html (hunk 1 above)
Comment (choiboii) [reply templates/acc_active_email.html]: Done!

File: users/models.py (hunk 2 above)
Comment (choiboii) [reply users/models.py:42]: I'm adding it so the user will be unable to login until they activate their account. In views.py, I have it so when the user is created, its is_active value is set to false, then it is set to true when they click the activation link in their email.

File: users/views.py (hunk 3 above)
Comment (choiboii) [reply users/views.py:15]: Yes! Encoding requires the the variable to be in byte form, and while after decoding, it needs to be reaffirmed into a string. I don't know the actual algorithm behind it, but it seems to implement python's encoding/decoding function with some other levels of further implementations with non-string characters. The source code is a little confusing to decipher.

**Review by jasonkang14 — APPROVED** [review]
LGTM

**Pull request merged**